core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2418,6 +2421,34 @@ public void testPaginationForListTables(int numberOfItems) {
             eq(ListTablesResponse.class));
   }
 
+  @Test
+  public void testReplaceTableKeepsSnapshotLog() {
```

Comment:
Could we put this test in `CatalogTests`? That'll cover both REST + non-REST cases (TestRESTCatalog extends `CatalogTests`).